infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -90,17 +130,55 @@ public long query(CorfuStoreMetadata.Timestamp timestamp, LogReplicationMetadata
         return val;
     }
 
+    /**
+     * Given a specific timestamp, get the current siteConfig metadata value.
+     * If the timestamp is null, get the most recent value.
+     * @param ts
+     * @return
+     */
+    public long getTopologyConfigId(CorfuStoreMetadata.Timestamp ts) {
+        return query(ts, LogReplicationMetadataType.TOPOLOGY_CONFIG_ID);
+    }
+
+    /**
+     * Get the most recent ConfigID.
+     * @return
+     */
     public long getTopologyConfigId() {
         return query(null, LogReplicationMetadataType.TOPOLOGY_CONFIG_ID);
     }
+    /**
+     * Given a specific timestamp, get the current version value.
+     * If the timestamp is null, get the most recent value.
+     * @param ts
+     * @return
+     */
+    public String getVersion(CorfuStoreMetadata.Timestamp ts) { return queryString(ts, LogReplicationMetadataType.VERSION); }
 
+    /**
+     * Get the most recent version.
+     * @return
+     */
     public String getVersion() { return queryString(null, LogReplicationMetadataType.VERSION); }
 
+    /**
+     * Given a specific timestamp, get the most recent started full snapshot sync's timestamp.
```

Comment:
Can you please clarify this? I still don't understand this.

Reply:
It is an UFO api. Please refer to: 
/**
     * Fetch the Value payload for the specified key at the specified snapshot/timestamp.
     *
     * @param tableName Table name.
     * @param timestamp Timestamp to perform the query.
     * @param key       Key.
     * @param <K>       Type of key.
     * @param <V>       Type of value/payload.
     * @param <M>       Type of metadata.
     * @return Value.
     */
    @Nullable
    public <K extends Message, V extends Message, M extends Message>
    CorfuRecord<V, M> getRecord(@Nonnull final String tableName,
                                @Nullable final Timestamp timestamp,
                                @Nonnull final K key) {

Reply:
But, we also have available, the one that fetches the latest value, without having to access the DB once more for a TS that is not really used, it's also a UFO API:

 /**
     * Fetch the CorfuRecord for the specified key at the LATEST timestamp (default).
     *
     * @param tableName Table name.
     * @param key       Key.
     * @param <K>       Type of key.
     * @param <V>       Type of the payload/value.
     * @param <M>       Type of the metadata.
     * @return CorfuRecord encapsulating the payload and the metadata.
     */
    @Nullable
    public <K extends Message, V extends Message, M extends Message>
    CorfuRecord<V, M> getRecord(@Nonnull final String tableName,
                                @Nonnull final K key)

Reply:
Because right now it is 5 different db calls, without giving timestamp, it can give an inconsistent view of the metadata. But anyway, in future we will change to one db call, the api will change without a timestamp needed.

Reply:
This will be addressed in https://github.com/CorfuDB/CorfuDB/issues/2626